core/kotlinx-coroutines-test/src/TestCoroutineExceptionHandler.kt

```diff
@@ -0,0 +1,54 @@
+package kotlinx.coroutines.test
+
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Access uncaught coroutine exceptions captured during test execution.
+ */
+@ExperimentalCoroutinesApi
+interface UncaughtExceptionCaptor {
+    /**
+     * List of uncaught coroutine exceptions.
+     *
+     * The returned list will be a copy of the currently caught exceptions.
+     *
+     * During [cleanupTestCoroutines] the first element of this list will be rethrown if it is not empty.
+     */
+    val uncaughtExceptions: List<Throwable>
+
+    /**
+     * Call after the test completes.
+     *
+     * @throws Throwable the first uncaught exception, if there are any uncaught exceptions
+     */
+    fun cleanupTestCoroutines()
+}
+
+/**
+ * An exception handler that can be used to capture uncaught exceptions in tests.
+ */
+@ExperimentalCoroutinesApi
+class TestCoroutineExceptionHandler: UncaughtExceptionCaptor, CoroutineExceptionHandler {
+
+    override fun handleException(context: CoroutineContext, exception: Throwable) {
+        synchronized(_exceptions) {
+            _exceptions += exception
+        }
+    }
+
+    override val key = CoroutineExceptionHandler
+
+    private val _exceptions = mutableListOf<Throwable>()
```

Comment:
i'm not familiar w/ kotlin src code but `_` seems unusual :man_shrugging:

Reply:
It seems to be a common behaviour when using a private backing field to represent a narrower public type, I'm to understand it's also the standard practice in some other languages, but sadly I can't provide a citation

Reply:
This is a private backing field  for a public property, _name is the pattern for that afaik

Reply:
Yes. Underscore is also used for atomics in our libraries code to represent "backing atomic" for a atomic value.